### PR TITLE
Parameterized graphql endpoint and caching

### PR DESCRIPTION
**Pull request by jihuang-adobe**
## Description

Instead of hardcoding it in usePersistedQueries.js, graphql endpoint namespace is now part of .env.
Graphql caching can be configured in .env.

## Related Issue

https://github.com/adobe/aem-guides-wknd-graphql/issues/95

## Motivation and Context

Both customers with AEM Guides WKND installed and Adobe internal with Reference Demo installed can use the code hosted on code sandbox, simply change the graphql endpoint namespace to enable the sample hosted app to pull content from their environment.
https://experienceleague.adobe.com/landing/experience-manager/headless/developer/code/basic-react-app.html?lang=en

## How Has This Been Tested?
tested locally and tested on online hosted sandbox
https://codesandbox.io/p/sandbox/wknd-react-app-forked-gfrtlf

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: react-app/src/api/usePersistedQueries.js [hunk 1]
```diff
@@ -70,16 +80,16 @@ export function useAdventuresByActivity(adventureActivity, params) {
...
         queryVariables = { ...queryVariables, activity: adventureActivity };
 
-        // Call the AEM GraphQL persisted query named "wknd-shared/adventures-by-activity" with parameters
-        response = await fetchPersistedQuery("wknd-shared/adventures-by-activity", queryVariables);
+        // Call the AEM GraphQL persisted query named "[graphql endpoint namspace]/adventures-by-activity" with parameters
+        response = await fetchPersistedQuery(REACT_APP_GRAPHQL_ENDPOINT + "/adventures-by-activity", queryVariables);
       } else {
-        // Call the AEM GraphQL persisted query named "wknd-shared/adventures-all"
-        response = await fetchPersistedQuery("wknd-shared/adventures-all", queryVariables);
+        // Call the AEM GraphQL persisted query named "[graphql endpoint namspace]/adventures-all"
```
Comment (davidjgonzalez) [inline react-app/src/api/usePersistedQueries.js:91]:
```suggestion
        // Call the AEM GraphQL persisted query named "[graphql endpoint namespace]/adventures-all"
```

File: react-app/src/api/usePersistedQueries.js [hunk 2]
```diff
@@ -98,7 +108,7 @@ export function useAdventuresByActivity(adventureActivity, params) {
 }
 
 /**
- * Calls the 'wknd-shared/adventure-by-slug' persisted query with `slug` parameter.
+ * Calls the '[graphql endpoint namspace]/adventure-by-slug' persisted query with `slug` parameter.
```
Comment (davidjgonzalez) [inline react-app/src/api/usePersistedQueries.js:111]:
```suggestion
 * Calls the '[graphql endpoint namespace]/adventure-by-slug' persisted query with `slug` parameter.
```

File: react-app/src/api/usePersistedQueries.js [hunk 3]
```diff
@@ -119,9 +129,9 @@ export function useAdventureBySlug(slugName, params) {
         slug: slugName,
       };
 
-      // Call the AEM GraphQL persisted query named "wknd-shared/adventure-by-slug" with parameters
+      // Call the AEM GraphQL persisted query named "[graphql endpoint namspace]/adventure-by-slug" with parameters
```
Comment (davidjgonzalez) [inline react-app/src/api/usePersistedQueries.js:132]:
```suggestion
      // Call the AEM GraphQL persisted query named "[graphql endpoint namespace]/adventure-by-slug" with parameters
```

**Review by davidjgonzalez — APPROVED** [review]
@jihuang-adobe can you take a look at these changes.

1. Changes the wknd-shared to be the default as afiak most people (outside Adobe SC) use wknd (and not RDA)
1. Changed semantics of the Cache toggle, since what youre really doing is disabling the cache w/ its addition (it caches by default). This also helps communicate that disabling is the "outlier" case. I also added a number of comments to re-emphasize disabling cache should not be done on Production.

File: react-app/.env.development [hunk 4]
```diff
@@ -6,9 +6,9 @@ REACT_APP_HOST_URI=https://publish-p123-e456.adobeaemcloud.com/
 # Graphql endpoint namespace
 REACT_APP_GRAPHQL_ENDPOINT=wknd-shared
 
-# CACHE and TTL can be configured as PersistedQuery header in AEM
-# The ability to disable CACHING is only for demo purposes
-ENABLE_CACHE=false
+# The ability to disable CACHING is ONLY for demo purposes. You should NEVER disable caching in production. Setting this to true will ad cache-busting parameters.
+# Production-use CACHE and TTL can be configured as PersistedQuery header in AEM
+DISABLE_CACHE=false
```
Comment (davidjgonzalez) [inline react-app/.env.development:11]:
```suggestion
# The ability to disable CACHING is ONLY for demo purposes. You should NEVER disable caching in production. Setting this to true will add cache-busting parameters to the AEM Headless API request.
# Production-use CACHE and TTL can be configured as PersistedQuery header in AEM
DISABLE_CACHE=false
```

**Comment by jihuang-adobe** [comment]
OMG, LGTM is good for me.  Thanks

**Pull request merged**